service/worker/deployment/deployment_workflow.go

```diff
@@ -128,6 +176,8 @@ func (d *DeploymentWorkflowRunner) run() error {
 	workflowArgs := &deployspb.DeploymentWorkflowArgs{
 		NamespaceName:     d.DeploymentLocalState.NamespaceName,
 		NamespaceId:       d.DeploymentLocalState.NamespaceId,
+		DeploymentName:    d.DeploymentLocalState.DeploymentName,
+		BuildId:           d.DeploymentLocalState.BuildId,
```

Comment:
again, I think this should be arranged so you can just pass `d.DeploymentLocalState` as a whole to the next run, and then you never have to worry about adding/removing fields here

Reply:
So instead of embedding args into state, adding state in args as field, right? that does seem safer to me. @Shivs11 I agree with David. Sorry if I pushed you to the wrong direction. let's just do it this way, similar to schedules.

Later if we want to do more optimization in the args, we can replace the `args.state` field with `args.compactedState` that holds compacted version of the state. But that's a worry for the future.

Reply:
ah, seemed like we did completed a full circle haha - no worries, thank you for the discussion nevertheless folks